packages/odata-service-writer/src/index.ts

```diff
@@ -88,26 +88,40 @@ async function generate(basePath: string, data: OdataService, fs?: Editor): Prom
 
         // copy existing `ui5.yaml` as starting point for ui5-mock.yaml
         const ui5MockConfig = await UI5Config.newInstance(ui5Config.toString());
-        ui5MockConfig.addMockServerMiddleware(data.path);
+        ui5MockConfig.addMockServerMiddleware(service.path);
         fs.write(join(basePath, 'ui5-mock.yaml'), ui5MockConfig.toString());
 
         // also add mockserver middleware to ui5-local.yaml
-        ui5LocalConfig.addMockServerMiddleware(data.path);
+        ui5LocalConfig.addMockServerMiddleware(service.path);
 
         // create local copy of metadata and annotations
-        fs.write(join(basePath, 'webapp', 'localService', 'metadata.xml'), prettifyXml(data.metadata, { indent: 4 }));
+        fs.write(
+            join(basePath, 'webapp', 'localService', 'metadata.xml'),
+            prettifyXml(service.metadata, { indent: 4 })
+        );
+
+        // Adds local annotations to datasources section of manifest.json and writes the annotations file
+        if (service.localAnnotationsName) {
+            const namespaces = getAnnotationNamespaces(service);
+            fs.copyTpl(
+                join(templateRoot, 'add', 'annotation.xml'),
+                join(basePath, 'webapp', 'annotations', `${service.localAnnotationsName}.xml`),
+                { ...service, namespaces }
```

Comment:
Do we use the `namespaces` in the `manifest.json` template?

Reply:
No, it's only used in the local annotation file.